contracts/security-token/ERC865.sol

```diff
@@ -0,0 +1,103 @@
+pragma solidity ^0.4.18;
+
+import "./CheckpointToken.sol";
+
+/* Largely copied from https://github.com/OpenZeppelin/openzeppelin-solidity/pull/741/files */
+
+contract ERC865 is CheckpointToken {
+  mapping(bytes => bool) signatures;
+
+  event TransferPreSigned(address indexed from, address indexed to, address indexed delegate, uint256 amount, uint256 fee);
+  event Debug(address from, bytes32 hash);
+
+  /**
+    * @notice Submit a presigned transfer
+    * @param _signature bytes The signature, issued by the owner.
+    * @param _to address The address which you want to transfer to.
+    * @param _value uint256 The amount of tokens to be transferred.
+    * @param _fee uint256 The amount of tokens paid to msg.sender, by the owner.
+    * @param _nonce uint256 Presigned transaction number
+    */
+  function transferPreSigned(
+    bytes _signature,
+    address _to,
+    uint256 _value,
+    uint256 _fee,
+    uint256 _nonce
+  )
+    public
+    returns (bool)
+  {
+    require(_to != address(0));
+    require(signatures[_signature] == false);
+    bytes32 hashedTx = transferPreSignedHashing(address(this), _to, _value, _fee, _nonce);
+    address from = recover(hashedTx, _signature);
+    require(from != address(0));
+
+    transferInternal(from, _to, _value);
+    transferInternal(from, msg.sender, _fee);
+
+    signatures[_signature] = true;
+    TransferPreSigned(from, _to, msg.sender, _value, _fee);
```

Comment:
Do we need to emit the standard `Transfer` event as well as most servers are only listening to ERC-20 `Transfer`?

Reply:
Added Transfer() for both, the fee and the actual payment.